### PR TITLE
Unit test for param files

### PR DESCRIPTION
**Pull request by herou**
I wrote unit tests required from this ticket:  #202

### CODE REVIEW

**Comment by gmale** [comment]
This looks like a good start :tada: 

From here, you probably want a setup function that runs before each unit test, that clears the param directory. Something along the lines of:

```kotlin
    @Before
    fun setup() {
        // clear the param files
        SaplingParamTool.clear(cacheDir)
    }
```
Now, each unit test is starting with an empty cache and is independent of other tests. Next, you would want to setup each test scenario:

```kotlin
    @Test
    fun testFilesExist(){
        // Given
        SaplingParamTool.fetchParams(cacheDir)
        
        // When
        val result = SaplingParamTool.validate(cacheDir)

        // Then
        Assert.assertFalse(result)
    }
 ```
 
If the `fetchParams` function works, then the previous test will succeed. So we can consider it tested. Meaning, we can build on it, as if it works, for the rest of the tests, without having to mock too much:

```kotlin
    @Test
    fun testOnlyOutputOFileExists(){
        // Given
        SaplingParamTool.fetchParams(cacheDir)
        File("$cacheDir/${ZcashSdk.SPEND_PARAM_FILE_NAME}").delete()
                
        // When
        val result = SaplingParamTool.validate(cacheDir)

        // Then
        Assert.assertFalse("Validation should fail when the spend params are missing", result)
    }
 ```

File: src/androidTest/java/cash/z/ecc/android/sdk/tool/SaplingParamToolTest.kt [hunk 1]
```diff
@@ -1,11 +1,56 @@
...
+
+    @Test
+    fun testFilesExist(){
+        val result = SaplingParamTool.validate(cacheDir)
+
+        Assert.assertFalse(result)
+    }
+
+    @Test
+    fun testOnlySpendFileExit(){
```
Comment (gmale) [inline src/androidTest/java/cash/z/ecc/android/sdk/tool/SaplingParamToolTest.kt:29]: minor typo `Exit` should be `Exists`

File: src/androidTest/java/cash/z/ecc/android/sdk/tool/SaplingParamToolTest.kt [hunk 2]
```diff
@@ -1,11 +1,56 @@
...
+        val result = SaplingParamTool.validate(cacheDir)
+
+        Assert.assertFalse(result)
+    }
+
+    @Test
+    fun testOnlyOutputOFileExit(){
```
Comment (gmale) [inline src/androidTest/java/cash/z/ecc/android/sdk/tool/SaplingParamToolTest.kt:36]: minor typo `Exit` should be `Exists`

**Review by gmale — CHANGES_REQUESTED** [review]

**Comment by herou** [comment]
@gmale I applied your changes above.

**Comment by gmale** [comment]
Discussed over telegram the `suspend` function issues and all tests pass. Merging. 

Nice work.

From here, the next steps are on my end.

**Pull request merged**